src/com/facebook/buck/jvm/kotlin/JarBackedReflectedKotlinc.java

```diff
@@ -157,7 +175,7 @@ private Object loadCompilerShim(ExecutionContext context) {
 
       ClassLoader classLoader =
           classLoaderCache.getClassLoaderForClassPath(
-              null /* parent classloader */,
+              ToolProvider.getSystemToolClassLoader(),
```

Comment:
Because `ToolProvider` itself is not thread safe, you'll need to synchronize this with the (currently single) other usage of `ToolProvider`, or else there may be strange and intermittent failures when running under `buckd`.

Reply:
Do you think that wrapping this inside a `synchronized (ToolProvider.class) { ... }` as `JdkProvidedInMemoryJavac` does is enough?

Reply:
That should do it. If you wouldn't mind, though, factor that out into a helper class somewhere so that we don't have to repeat the comment every time.